src/lint/track_config.nim

```diff
@@ -621,6 +621,44 @@ proc satisfiesSecondPass(trackConfigContents: string; path: Path): bool =
   checkExercisesPCP(practiceExercises, result, path)
   checkExerciseSlugsAndForegone(exercises, result, path)
 
+proc getExerciseSlugs(data: JsonNode; k: string): HashSet[string] =
+  result = initHashSet[string]()
+  if data.kind != JObject or "exercises" notin data:
+    return
+
+  let exercises = data["exercises"]
+
+  if exercises.kind != JObject or k notin exercises or exercises[k].kind != JArray:
+    return
+
+  for exercise in exercises[k]:
+    if exercise.kind == JObject:
+      if exercise.hasKey("slug"):
+        let slug = exercise["slug"]
+        if slug.kind == JString:
+          let slugStr = slug.getStr()
+          if slugStr.len > 0:
+            result.incl slugStr
+
+proc checkExerciseDirsAreInTrackConfig(trackDir: Path; data: JsonNode;
+                                       b: var bool; path: Path) =
+  ## Sets `b` to `false` if there is an exercise directory that is
+  ## not an exercise `slug` in `data`.
+  for exerciseKind in ["concept", "practice"]:
+    let exerciseSlugs = getExerciseSlugs(data, exerciseKind)
+    if exerciseSlugs.len > 0:
+      let exercisesDir = trackDir / "exercises" / exerciseKind
+      if dirExists(exercisesDir):
+        for exerciseDir in getSortedSubdirs(exercisesDir):
+          let dirSlug = lastPathPart(exerciseDir.string)
+          if dirSlug notin exerciseSlugs:
+            let msg = &"{q $exercisesDir} contains a directory named {q dirSlug}, " &
+                      &"which is not a `slug` in the array of {exerciseKind} " &
+                       "exercises. Please add the exercise to that array. " &
+                       "If the exercise is not ready to be shown on the " &
+                       "website, please set its `status` value to \"wip\""
```

Comment:
Comment: there's an ongoing question about how to refer to JSON key/value pairs. Here it feels painful to write `wip` with backticks like:

> please set its \`status\` value to \`wip\`

rather than with double quotes:

> please set its \`status\` value to \"wip\"

But that raises the question about whether we should write \"status\" with double quotes too (which sometimes feels worse in other error messages).